src/mscorlib/src/System/AppDomainUnloadedException.cs

```diff
@@ -16,6 +16,8 @@
 
 namespace System
 {
+    [Serializable]
+    [System.Runtime.CompilerServices.TypeForwardedFrom("mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089")]
     internal class AppDomainUnloadedException : SystemException
```

Comment:
Note that the public AppDomainUnloadedException lives in CoreFX. This is a dead copy that we have not cleaned up yet.

Reply:
We never throw this either so we can certainly leave it unserializable.

Reply:
In another change some point it can be removed along with ThreadAbortException I think.

Reply:
(ThreadAbortException can't be removed. It is used internally by the debugger funceval.)

Reply:
My bad, I didn't search vm. I see there are places there that could throw ADUE also, but perhaps they will never execute.

Reply:
If we throw it in native code somewhere it needs to stay serializable and also needs the TypeForwardedFrom attribute so that the BinaryFormatter replaces the AssemblyQualifiedName (System.Private.Corelib) with mscorlib when serializing it. GenAPI automatically typeforwards to the public type in corefx so everything should be fine